src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/alignment/AssemblyContigAlignmentsConfigPicker.java

```diff
@@ -266,13 +266,62 @@ private static double computeTigExplainQualOfOneAlignment(final List<AlignmentIn
         return numFirst.thenComparing(mismatchSecond);
     }
 
-    private static List<AlignmentInterval> splitGaps(final List<AlignmentInterval> configuration) {
-        return configuration.stream()
+    @VisibleForTesting
+    static List<AlignmentInterval> splitGaps(final List<AlignmentInterval> configuration) {
+
+        final List<Integer> blackList = new ArrayList<>(configuration.size());
```

Comment:
It seems inefficient to do the splitting twice. I'm also not a huge fan of this blacklist list-of-indices. How about a different approach, something like:

1) Make a map of input AI's that can be split to their split forms.
2) Make a copy of the input AI list to hold the results.
3) Check each of the split alignments from 1) against the input list. Replace the entries in the output list corresponding to their input alignments, and remove any entries from the output list that fail as a result of these checks.

Would that work?

Reply:
changed to am implementation where a list holds the split alignments (or original if no large gaps), for alignments that fails the check, its corresponding split is invalidated (set to null), then the list is collected on those whose split is still valid after the filter.

Just out of curiosity: are you concerned more about the blacklist approach in general or the fact that the blacklist was storing indices?